### PR TITLE
Include css class name in compiled output

### LINKED ISSUE #115: Mangle names only for final builds
Development server should not mangle names so it's easier to see Dom structure in browser's inspector

### PR DESCRIPTION
**Pull request by gdotdesign**
Supersedes #446 with original commits intact and resolves #115

### CODE REVIEW

**Review by matthewmcgarvey — APPROVED** [review]

File: src/style_builder.cr [hunk 1]
```diff
@@ -192,14 +214,14 @@ module Mint
...
 
-    def prefixed_class_name(node)
-      @css_prefix.to_s + style_pool.of(node, nil)
+    def prefixed_class_name(node : Ast::Style, id : String? = nil)
+      @css_prefix.to_s + style_pool.of(node, id)
     end
 
     # The main entry point for processing a "style" tag.
-    def process(node : Ast::Style)
+    def process(node : Ast::Style, id : String)
```
Comment (Sija) [inline src/style_builder.cr:222]: Shouldn't this allow for nil-able `id` too?

```suggestion
    def process(node : Ast::Style, id : String? = nil)
```

File: src/style_builder.cr [hunk 2]
```diff
@@ -17,6 +17,28 @@ module Mint
...
+
+    def of(subject : Ast::Style, id : String? = nil)
+      if @optimize
+        @pool.of(subject, nil)
+      else
+        @cache[subject] ||= begin
+          if id
+            "#{id}_#{subject.name.value.gsub('-', '_')}"
+          else
+            subject.name.value
```
Comment (Sija) [inline src/style_builder.cr:35]: Shouldn't this be escaped too?

```suggestion
            subject.name.value.gsub('-', '_')
```

Comment (gdotdesign) [reply src/style_builder.cr:35]: I don't think so, why?

Comment (Sija) [reply src/style_builder.cr:35]: In the same vein as the line above I guess.

Comment (gdotdesign) [reply src/style_builder.cr:35]: Oh right, I was thinking you commented on *that* line :joy:

File: src/style_builder.cr (hunk 1 above)
Comment (gdotdesign) [reply src/style_builder.cr:222]: It compiles so _I guess not_?

Comment (Sija) [reply src/style_builder.cr:222]: I'm asking since the API all around it allows for empty `id`, so I don't see a reason why this one shouldn't.

Comment (Sija) [reply src/style_builder.cr:222]: It calls `prefixed_class_name` which has nil-able `id`, this, in turn, invokes `StylePool#of` which does the same, etc.

Comment (Sija) [reply src/style_builder.cr:222]: > It compiles so _I guess not_?

Just because there are no tests stressing this path :P

**Pull request merged**